cpp/src/join/join_common_utils.cuh

```diff
@@ -322,6 +331,32 @@ __device__ void flush_output_cache(unsigned int const activemask,
   }
 }
 
+template <int num_warps, cudf::size_type output_cache_size>
+__device__ void flush_left_cache(unsigned int const activemask,
```

Comment:
Don't need to add new function, can also just use overloads or default parameters. IDK what the preferences are for libcudf so I just kept it simple.

Reply:
As you mentioned, building semi/anti join is quite expensive thus adding overloads is probably the right way to go.